plugins/dql-backend/src/service/queries.ts

```diff
@@ -32,7 +32,7 @@ export const dynatraceQueries: Record<string, string | undefined> = {
     | fieldsAdd Problems=coalesce(Problems,0)
     | lookup [ fetch events, from: -30m | filter event.kind=="SECURITY_EVENT" | filter event.category=="VULNERABILITY_MANAGEMENT" | filter event.provider=="Dynatrace" | filter event.type=="VULNERABILITY_STATE_REPORT_EVENT" | filter in(vulnerability.stack,{"CODE_LIBRARY","SOFTWARE","CONTAINER_ORCHESTRATION"}) | filter event.level=="ENTITY" | summarize { workloadId=arrayFirst(takeFirst(related_entities.kubernetes_workloads.ids)), vulnerability.stack=takeFirst(vulnerability.stack)}, by: {vulnerability.id, affected_entity.id} | summarize { Vulnerabilities=count() }, by: {workloadId}], sourceField:id, lookupField:workloadId, fields:{Vulnerabilities}
     | fieldsAdd Vulnerabilities=coalesce(Vulnerabilities,0)
-    | filter workload.labels[\`backstage.io/component\`] == "\${componentNamespace}.\${componentName}"
+    \${additionalFilter}
```

Comment:
I would suggest to add the logic of building the filters into the query.

Reply:
Also the labels would be a bit hard to parse in the query.
e.g., `backstage.io/kubernetes-label-selector: stage=hardening,name=frontend` should result in ``| filter workloard[`stage`] == "hardening" AND workloard[`name`] == "frontend" ``

Reply:
Ok, now I understood it. I moved the logic to the query file👍